### ISSUE TITLE
DateTimeIndex.to_frame() drops timezone in pandas 0.24

### ISSUE THREAD

**Issue by jakevdp** [pandas-dev/pandas#25809]
Behavior in pandas 0.23:
```python
>>> import pandas as pd
>>> pd.__version__
0.23.0
>>> x = pd.date_range(start='2019-01-01', end='2019-01-30', freq='D')
>>> x = x.tz_localize('UTC')
>>> x.to_frame().dtypes
0    datetime64[ns, UTC]
dtype: object
```

Behavior in pandas 0.24

```python
>>> import pandas as pd
>>> pd.__version__
0.24.2
>>> x = pd.date_range(start='2019-01-01', end='2019-01-30', freq='D')
>>> x = x.tz_localize('UTC')
>>> x.to_frame().dtypes
0    datetime64[ns]
dtype: object
```

**Comment by jakevdp** [comment]
Issue is here: https://github.com/pandas-dev/pandas/blob/05780f864f01653d4d7791cd4df66aca54716b4e/pandas/core/indexes/base.py#L1211

index.values converts to a numpy array with no timezone information.

**Comment by jakevdp** [comment]
Some initial experiments suggest that replacing ``self.values`` with ``self.to_numpy()`` will fix this; I'll give this a try and prepare a PR.

**Comment by mroeschke** [comment]
Thanks for the report @jakevdp! I already have a PR up to fix this issue #25811. 

The semantics with `.values` changed recently along with our migration to use ExtensionArrays more internally (in this case `DatetimeArray`). Passing the internal extension array (`._values`) into the constructor solves this issue.
